src/browser/BrowserService.cpp

```diff
@@ -58,34 +60,45 @@ const QString BrowserService::OPTION_ONLY_HTTP_AUTH = QStringLiteral("BrowserOnl
 // Multiple URL's
 const QString BrowserService::ADDITIONAL_URL = QStringLiteral("KP2A_URL");
 
-BrowserService::BrowserService(DatabaseTabWidget* parent)
-    : m_dbTabWidget(parent)
+Q_GLOBAL_STATIC(BrowserService, s_browserService);
+
+BrowserService::BrowserService()
+    : QObject()
+    , m_browserHost(new BrowserHost)
     , m_dialogActive(false)
     , m_bringToFrontRequested(false)
     , m_prevWindowState(WindowState::Normal)
     , m_keepassBrowserUUID(Tools::hexToUuid("de887cc3036343b8974b5911b8816224"))
 {
-    // Don't connect the signals when used from DatabaseSettingsWidgetBrowser (parent is nullptr)
-    if (m_dbTabWidget) {
-        connect(m_dbTabWidget, SIGNAL(databaseLocked(DatabaseWidget*)), this, SLOT(databaseLocked(DatabaseWidget*)));
-        connect(
-            m_dbTabWidget, SIGNAL(databaseUnlocked(DatabaseWidget*)), this, SLOT(databaseUnlocked(DatabaseWidget*)));
-        connect(m_dbTabWidget,
-                SIGNAL(activateDatabaseChanged(DatabaseWidget*)),
-                this,
-                SLOT(activateDatabaseChanged(DatabaseWidget*)));
+    connect(m_browserHost, &BrowserHost::clientMessageReceived, this, &BrowserService::processClientMessage);
+    setEnabled(browserSettings()->isEnabled());
+}
+
+BrowserService* BrowserService::instance()
+{
+    return s_browserService;
+}
+
+void BrowserService::setEnabled(bool state)
```

Comment:
Maybe just `enabled` instead of `state` because it's a boolean, not a possible multi-value state.